modules/core/04-channel/keeper/packet_test.go

```diff
@@ -940,3 +988,74 @@ func (suite *KeeperTestSuite) TestAcknowledgePacket() {
 		})
 	}
 }
+
+// TestAcknowledgeFlush tests that Acknowledging the last in-flight packet moves the channel
+// state to FLUSHINGCOMPLETE.
+func (suite *KeeperTestSuite) TestAcknowledgeFlushStatus() {
+	var path *ibctesting.Path
+
+	testCases := []struct {
+		msg                 string
+		malleate            func()
+		expectedFlushStatus types.FlushStatus
+	}{
+		{
+			"success: final packet commitment cleared, flush status set to FLUSHCOMPLETE",
+			func() {},
+			types.FLUSHCOMPLETE,
+		},
+		{
+			"success: has in-flight packets, flush status remains FLUSHING",
+			func() {
+				// Send an additional packet
+				sequence, err := path.EndpointA.SendPacket(defaultTimeoutHeight, disabledTimeoutTimestamp, ibctesting.MockPacketData)
+				suite.Require().Equal(uint64(1), sequence)
+				suite.Require().NoError(err)
+			},
+			types.FLUSHING,
+		},
+	}
+
+	for i, tc := range testCases {
+		tc := tc
+		suite.Run(fmt.Sprintf("Case %s, %d/%d tests", tc.msg, i, len(testCases)), func() {
+			suite.SetupTest() // reset
+			path = ibctesting.NewPath(suite.chainA, suite.chainB)
+			suite.coordinator.Setup(path)
+
+			tc.malleate()
```

Comment:
Just a nit, but maybe it would make more sense to me to call `tc.malleate()` after receiving the packet on chain B.

Reply:
yea, makes sense to me too